src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1051,6 +1061,52 @@ private void RegisterHomeWorkspace(HomeWorkspaceModel newWorkspace)
 
         #endregion
 
+        #region backup/timer
+
+        /// <summary>
+        /// Backup all the files
+        /// </summary>
+        protected void SaveBackupFiles(object state)
+        {
+            foreach (var workspace in Workspaces)
+            {
+                if (!workspace.HasUnsavedChanges)
+                    continue;
+
+                string fileName;
+                if (string.IsNullOrEmpty(workspace.FileName))
+                {
+                    fileName = "recent_new_file_" + workspace.Guid;
+                    var ext = workspace is HomeWorkspaceModel ? ".DYN" : ".DYF";
+                    fileName += ext;
```

Comment:
I am taking the first point. But for the point related to the prefix, if FileName of the workspace is not valid, it means it has never been saved, so "recent_new_file_" makes more sense.